massa-execution-worker/src/execution.rs

```diff
@@ -241,6 +242,26 @@ impl ExecutionState {
             .set_active_cursor(self.active_cursor.period, self.active_cursor.thread);
         self.massa_metrics
             .set_final_cursor(self.final_cursor.period, self.final_cursor.thread);
+
+        self.massa_metrics.inc_operations_final_counter(
+            exec_out_2.state_changes.executed_ops_changes.len() as u64,
+        );
+
+        // Broadcast a final slot execution output to active channel subscribers.
+        if self.config.broadcast_enabled {
+            let slot_exec_out = SlotExecutionOutput::FinalizedSlot(exec_out_2);
+            if let Err(err) = self
+                .channels
+                .slot_execution_output_sender
+                .send(slot_exec_out)
+            {
+                trace!(
+                    "error, failed to broadcast final execution output for slot {} due to: {}",
+                    exec_out.slot,
+                    err
+                );
+            }
+        }
```

Comment:
move those lines up to replace `let exec_out_2 = exec_out.clone();` and avoid cloning the whole state changes if broadcast is not enabled (only clone in the `if self.config.broadcast_enabled`)